### PR TITLE
vtol_att_control: handle front transition timeout for all types

### PR DESCRIPTION
**Pull request by RomanBapst**
- moved front transition timout logic to base class so that it's available
for all vtol types
- cleaned up variable usage

Signed-off-by: Roman <bapstroman@gmail.com>

### CODE REVIEW

**Comment by RomanBapst** [comment]
@dagar This moves front transition timeout logic to base class so that it is available for all vtol types.
It also contains a cleanup of variables.

**Comment by dagar** [comment]
That looks much better. What do you think about back transition timeout? 

Rebase on master to get CI working (https://github.com/PX4/Firmware/pull/9339).

File: src/modules/vtol_att_control/standard.cpp [hunk 1]
```diff
@@ -323,7 +299,7 @@ void Standard::update_mc_state()
 
 	// Do not engage pusher assist during a failsafe event
 	// There could be a problem with the fixed wing drive
-	if (_attc->get_vtol_vehicle_status()->vtol_transition_failsafe) {
+	if (_attc->get_vtol_vehicle_status().vtol_transition_failsafe) {
```
Comment (RomanBapst) [inline src/modules/vtol_att_control/standard.cpp]: @dagar How about _vtol_vehicle_status.vtol_transition_failsafe ?

**Comment by RomanBapst** [comment]
@dagar I combined this PR with your cleanup PR #9352 
I'll also give this a test on the tiltrotor.

**Pull request closed without merging**